src/vec/mod.rs

```diff
@@ -16,6 +16,18 @@ use crate::storage::{OwnedStorage, Storage, ViewStorage};
 mod drain;
 pub use drain::Drain;
 
+/// The error type for fallible [`Vec`] and [`String`](crate::String) methods.
+#[derive(Debug)]
+pub struct CapacityError;
```

Comment:
This should be at the top level of the crate, given it's shared by all types.

Reply:
Also, make it `#[non_exhaustive]` just in case.

Reply:
Moved to the top level.

I didn't use an enum because you can't create a value of that type.

Reply:
I did not know that, thanks, fixed!